examples/index.html

```diff
@@ -9,12 +9,30 @@
       window.addEventListener('load', function () {
         const iframe = document.getElementById('exampleFrame');
         const nav = document.querySelector('nav');
+
+        function isMobile () {
+          return !(nav.getBoundingClientRect().width < document.body.getBoundingClientRect().width / 2);
+        }
+
+        // Initial iframe src based on URL search parameter
+        const queryMatch = location.search.match(/example=[^&]+/g);
```

Comment:
Could we potentially match on anchors? Say `www.links-lang.org/examples#dictSuggest.links`.

Reply:
for obivous reasons it can't just be the name of the Links file but must be the full path, but yeah, it now uses `location.hash`

Reply:
I see. It is not behaving as I expected. Moving back and forth in the browser history do not update the displayed content. I suppose this is due to the `iframe`. Maybe what I really want is to rid of the `iframe`, and use a method similar to YouTube and Twitter for rendering.

Reply:
`examples#/examples/dictionary/dictSuggest.links` do not read as nice as `examples#dictionary/dictSuggest.links`. So if we cannot get rid of the `/examples/` then I probably do not want to use anchors.

Reply:
Twitter, YouTube etc. have the advantage of knowing what their content looks like.
Since our examples are pretty much arbitrary we have to use an `iframe` (unless we make an AJAX request, parse the HTML and construct the DOM tree locally, but really... no!).

It is possible to emulate some of what you want with `onpopstate` etc. but that will take a little longer than 5 minutes ;)

Regarding the slash: We can of course trim/append the leading slash but then we would make very implicit assumtions about the `href` attribute.

An alternative is to just set the URL to the example page. Disadvantage here is, that when you reload the page/share the URL you will end up on the page with _only_ the example and no side-navigation etc.

Reply:
> Regarding the slash: We can of course trim/append the leading slash but then we would make very implicit assumtions about the href attribute.

I think it is OK make such assumptions as this is a thing intrinsic to the examples page. Personally, I dislike urls with query parameters because they leak implementation details. The reason I like the anchor is that `links-lang.org/examples#progress.links` reads like record projection :-). The examples page is essential one big record.

> It is possible to emulate some of what you want with onpopstate etc. but that will take a little longer than 5 minutes ;)

We should really have navigation on the examples page working properly.

> Twitter, YouTube etc. have the advantage of knowing what their content looks like.
Since our examples are pretty much arbitrary we have to use an iframe (unless we make an AJAX request, parse the HTML and construct the DOM tree locally, but really... no!).

Thanks for the clarification.

Reply:
Turns out `iframe` and `history` don't go well together.

Iframe-loading adds an entry to the browser history (which is not editable).
Adding a hash (`loaction.hash = `, `replaceState`, `pushState`) adds an entry to the browser history.
So when we update both there are two new entries in the browser history for the single user-action. 
When the user then starts navigating via back and forward there every other state has hash and iframe out of sync.

Reply:
> Adding a hash (loaction.hash =, replaceState, pushState) adds an entry to the browser history.

I take this would not be a problem with query parameters?

Reply:
unfortunately no. Only real difference is that with hash I have the benefit of the `onhashchange` event to work with. Otherwise they are pretty much the same.